File: tests/java-rosettacode/Doors100.java
public class Doors100 {
    public static void main(String[] args) {
        boolean[] doors = new boolean[101];
        for (int pass = 1; pass <= 100; pass++) {
            for (int door = pass; door <= 100; door += pass) {
                doors[door] = !doors[door];
            }
        }
        for (int i = 1; i <= 100; i++) {
            if (doors[i]) System.out.printf("Door %d is open%n", i);
        }
    }
}

File: tests/rosettacode/doors100.go
package main

import "fmt"

func main() {
	var doors [101]bool
	for pass := 1; pass <= 100; pass++ {
		for door := pass; door <= 100; door += pass {
			doors[door] = !doors[door]
		}
	}
	fmt.Print("Open doors:")
	for i := 1; i <= 100; i++ {
		if doors[i] {
			fmt.Printf(" %d", i)
		}
	}
	fmt.Println()
}
